### DISCUSSION TITLE
async Responder impl in Rocket 0.5

### DISCUSSION THREAD

**Discussion by barlindhaug** [rwf2/Rocket#1727] [Questions] [answered]
I have a responder that needs to call an async function, here `foo`, Is this possible in Rocket 0.5?

```
#[rocket::async_trait]
impl<'r> Responder<'r, 'static> for Error {
    async fn respond_to(self, req: &'r Request<'_>) -> response::Result<'static> {
        foo().await
    }
}
```

Will give the following error:

```
error[E0195]: lifetime parameters or bounds on method `respond_to` do not match the trait declaration
async fn respond_to(self, req: &'r Request<'_>) -> response::Result<'static> {
lifetimes do not match method in trait
```

**Comment by SergioBenitez** [comment, marked as answer]
It is not. You should have a constructor that is `async` instead. See, for example, `NamedFile::open()`.

**Reply by barlindhaug** [reply]
Thank you for your quick reply.

I also see this is already answered here for anyone interested:
https://github.com/SergioBenitez/Rocket/issues/1065#issuecomment-811182735
